src/Integration.UnitTests/ProfileConflicts/ConflictsManagerTests.cs

```diff
@@ -0,0 +1,378 @@
+                this.fileSystem.RegisterFile(solutionRuleSet);
+            }
+        }
+
+        private IEnumerable<RuleSetDeclaration> SetSingleValidRuleSetDeclarationPerProject(string configurationName = "Configuration", bool useUniqueProjectRuleSets = false)
+        {
+            List<RuleSetDeclaration> declarations = new List<RuleSetDeclaration>();
+
+            foreach (ProjectMock project in this.projectHelper.Projects.OfType<ProjectMock>())
+            {
+                var configuration = new ConfigurationMock(configurationName);
+                project.ConfigurationManager.Configurations.Add(configuration);
+
+                PropertyMock ruleSetProperty = configuration.Properties.RegisterKnownProperty(Constants.CodeAnalysisRuleSetPropertyKey);
+                ruleSetProperty.Value = project.FilePath.ToUpperInvariant(); // Catch cases where file paths are compared without OrdinalIgnoreCase 
+                if (useUniqueProjectRuleSets)
+                {
+                    ruleSetProperty.Value = Path.ChangeExtension(project.FilePath, configurationName + ".ruleSet");
+                }
+
+                PropertyMock ruleSetDirectories = configuration.Properties.RegisterKnownProperty(Constants.CodeAnalysisRuleSetDirectoriesPropertyKey);
+                ruleSetDirectories.Value = ConflictsManager.CombineDirectories(new[] { this.TestContext.TestRunDirectory, "." });
+
+                RuleSetDeclaration declaration = new RuleSetDeclaration(
+                    project,
+                    ruleSetProperty,
+                    ruleSetProperty.Value.ToString().ToLowerInvariant(), // Catch cases where file paths are compared without OrdinalIgnoreCase
+                    configuration.ConfigurationName,
+                    ruleSetDirectories.Value.ToString().Split(';'));
+
+                declarations.Add(declaration);
+
+                this.ruleSetInfoProvider.RegisterProjectInfo(project, declaration);
+            }
+
+            return declarations;
+        }
+
+        private IEnumerable<RuleSetDeclaration> SetValidRuleSetDeclarationPerProjectAndConfiguration(bool useUniqueProjectRuleSets, params string[] configurationNames)
+        {
+            List<RuleSetDeclaration> declarations = new List<RuleSetDeclaration>();
+
+            foreach (string configurationName in configurationNames)
+            {
+                declarations.AddRange(this.SetSingleValidRuleSetDeclarationPerProject(configurationName, useUniqueProjectRuleSets));
+            }
+
+            return declarations;
+        }
+
+        private void ConigureInspectorWithConflicts(IEnumerable<RuleSetDeclaration> knownDeclarations, Func<RuleConflictInfo> conflictFactory)
+        {
+            this.inspector.FindConflictingRulesAction = (baseline, ruleset, directories) =>
+            {
+                Assert.IsNotNull(baseline);
```

Comment:
Presumably the other parameters shouldn't be null either.
(Typo in the method name)